### PR TITLE
Feature/my reading saved

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制] 我的閱讀 - 收藏頁](https://app.asana.com/0/1203699264568808/1206168671896409/f)

# Notice

# Dependency

### CODE REVIEW

File: src/components/my-reading/saved-bookmarks.js [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+        toastr({ text: '已取消收藏' })
+        setBookmarks(prevBookmarks =>
+          _.filter(prevBookmarks, bookmark => bookmark.id !== bookmarkID)
+        )
+        setTotalSavedBookmarkCount(
+          prevTotalSavedBookmark => prevTotalSavedBookmark - 1
+        )
+      })
+      .catch(error => {
+        console.error('error: ', error)
```
Comment (Aylie-Chou) [inline src/components/my-reading/saved-bookmarks.js:118]: Question: what is the use of this console log ?

File: src/components/my-reading/saved-bookmarks.js [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+  const { removeAction } = useBookmark(store)
+  const { showSnackBar, snackBarText, toastr } = useSnackBar()
+  const navigate = useHistory()
+
+  const removeBookmark = bookmarkID => {
+    removeAction(bookmarkID)
+      .then(() => {
+        toastr({ text: '已取消收藏' })
+        setBookmarks(prevBookmarks =>
+          _.filter(prevBookmarks, bookmark => bookmark.id !== bookmarkID)
```
Comment (Aylie-Chou) [inline src/components/my-reading/saved-bookmarks.js:111]: (nice to have) we could use [`remove`](https://lodash.com/docs/4.17.15#remove) to make it easier to understand

File: src/components/my-reading/saved-bookmarks.js [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+      hero_image: {
+        resized_targets: {
+          mobile: {
+            url: thumbnail,
+          },
+        },
+      },
+      category,
+      category_set: [{ category: { name: category } }],
+      published_date: Number(`${publishedDate}000`),
```
Comment (Aylie-Chou) [inline src/components/my-reading/saved-bookmarks.js:149]: Question: why do we need to add `000` at the end of publishedDate

File: src/components/my-reading/saved-bookmarks.js [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+}
+
+SavedBookmarks.propTypes = {
+  isDeleting: PropTypes.bool,
+  isFetching: PropTypes.bool,
+  jwt: PropTypes.string,
+  userID: PropTypes.number,
+  getMultipleBookmarks: PropTypes.func,
+  page: PropTypes.number,
+  totalPages: PropTypes.number,
```
Comment (Aylie-Chou) [inline src/components/my-reading/saved-bookmarks.js:253]: Could we mark `isRequired` for props which should always have values

File: src/containers/MyReading.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        />
+      )
+    }
+    return (
+      <div>
+        <Title2
+          title={'已收藏'}
+          renderButton={size => moreSavedBookmarkBtn(size)}
+        />
+        <EmptyBox>空無一物</EmptyBox>
```
Comment (Aylie-Chou) [inline src/containers/MyReading.js:113]: 😆

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
Please have a look at comments ~~

File: src/components/my-reading/saved-bookmarks.js (hunk 3 above)
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js:149]: cause `date2yyyymmdd` only accept `UNIX timestamp in mileseconds`

File: src/components/my-reading/browsing-history.js [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+          </SnackBarContainer>
+        </>
+      )}
+    </Container>
+  )
+}
+
+BrowsingHistory.propTypes = {
+  page: PropTypes.number,
+  totalPages: PropTypes.number,
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply src/components/my-reading/browsing-history.js:264]: fixed

File: src/components/my-reading/saved-bookmarks.js (hunk 2 above)
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js:111]: fixed

File: src/components/my-reading/saved-bookmarks.js [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    false
+  )
+  const isFetching = _.get(
+    state,
+    [reduxStateFields.bookmarks, 'isRequesting'],
+    false
+  )
+  const jwt = _.get(state, [reduxStateFields.auth, 'accessToken'])
+  const userID = _.get(state, [reduxStateFields.auth, 'userInfo', 'user_id'])
+  const isAuthed = _.get(state, [reduxStateFields.auth, 'isAuthed'])
```
[the remark this answers is not in the document]
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js:288]: fixed

File: src/components/my-reading/saved-bookmarks.js (hunk 4 above)
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js:253]: fixed

File: src/containers/MyReading.js (hunk 5 above)
Comment (liruchen32) [reply src/containers/MyReading.js:113]: ⭕️

File: src/components/my-reading/saved-bookmarks.js (hunk 4 above)
Comment (Aylie-Chou) [reply src/components/my-reading/saved-bookmarks.js]: also `getMultipleBookmarks`, `isDeleting`, `isFetching` ?

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
almost there !!

File: src/components/my-reading/saved-bookmarks.js (hunk 2 above)
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js]: fixed

File: src/components/my-reading/saved-bookmarks.js (hunk 4 above)
Comment (liruchen32) [reply src/components/my-reading/saved-bookmarks.js]: fixed

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**